arangod/Aql/AqlValue.cpp

```diff
@@ -334,21 +334,46 @@ bool AqlValue::isArray() const noexcept {
   return true;
 }
 
-char const* AqlValue::getTypeString() const noexcept {
-  if (isNone()) {
-    return "none";
-  } else if (isNull(true)) {
-    return "null";
-  } else if (isBoolean()) {
-    return "bool";
-  } else if (isNumber()) {
-    return "number";
-  } else if (isString()) {
-    return "string";
-  } else if (isObject()) {
-    return "object";
-  } else if (isArray()) {
-    return "array";
+std::string_view AqlValue::getTypeString() const noexcept {
+  auto get = [](auto t) -> std::string_view {
+    switch (t) {
+      case velocypack::ValueType::Null:
+        return "null";
+      case velocypack::ValueType::Bool:
+        return "bool";
+      case velocypack::ValueType::Array:
+        return "array";
+      case velocypack::ValueType::Object:
+        return "object";
+      case velocypack::ValueType::Double:
+      case velocypack::ValueType::Int:
+      case velocypack::ValueType::UInt:
+      case velocypack::ValueType::SmallInt:
+        return "number";
+      case velocypack::ValueType::String:
+        return "string";
+      default:
+        return "none";
```

Comment:
The behavior now is not equivalent to the previous behavior before the change.
Note that before this change, `"null"` was returned even for None values, because of the following code part:
```
} else if (isNull(true)) {
    return "null";
...
```
`isNull(true)` returns `true` for real Null values, but also for None values.

Reply:
Yes, but it already checked none above, so no difference?

Reply:
Turns out that my comment does not matter. The old code checked for `Slice::isNone()` first and then returned `"none"` anyway.